drivers/scheduler/k8s/k8s.go

```diff
@@ -3153,16 +3153,20 @@ func (k *K8s) Destroy(ctx *scheduler.Context, opts map[string]bool) error {
 	var podList []corev1.Pod
 
 	allRules, err := k8sAutopilot.ListAutopilotRules()
-	if err != nil {
-		return err
-	}
-
-	for _, rule := range allRules.Items {
-		log.InfoD("Cleaning up autopilot rule: %s", rule.Name)
-		err := k8sAutopilot.DeleteAutopilotRule(rule.Name)
-		if err != nil {
-			return err
+	if err == nil {
```

Comment:
Few comments here:

- This function `Destroy` is meant only to delete the kubernetes objects which got deployed during the app creation
- As you can see in all the lines below, we are looping over SpecList and destroying them by explicitly calling the relevant destroy function
- Since autopilot rule is not getting created as part of any spec, it should be a separate function altogether. I would suggest you create a function explicitly in `common.go` to delete the autopilot rules and call it in the test case wherever needed
- The whole logic of deleting autopilot rule does not fit here because we are not using the receiver type `k *K8s` neither are we using the input arguments `(ctx *scheduler.Context, opts map[string]bool)`